rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll

```diff
@@ -366,6 +387,29 @@ module Impl {
     override string getAPrimaryQlClass() { result = "VariableAccess" }
   }
 
+  /** Holds if `e` occurs in the LHS of a (compound) assignment. */
+  private predicate assignLhs(Expr e) {
+    exists(BinaryExpr be |
+      be.getOperatorName().regexpMatch(".*=") and
+      e = be.getLhs()
+    )
+    or
+    exists(Expr mid |
+      assignLhs(mid) and
+      getImmediateParent(e) = mid
+    )
```

Comment:
What situation is this case intended to cover?

Reply:
Destructuring assignments: https://doc.rust-lang.org/reference/expressions/operator-expr.html#destructuring-assignments